In [1]:
#import all needed libraries
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
iris= load_iris()
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np

In [2]:
#display methods of iris
dir(iris)

['DESCR',
 'data',
 'data_module',
 'feature_names',
 'filename',
 'frame',
 'target',
 'target_names']

In [3]:
#logistic regression
lr_scores= cross_val_score(LogisticRegression(), iris.data, iris.target)
lr_scores

E:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

In [4]:
np.average(lr_scores)

0.9733333333333334

In [5]:
#decision tree
dtree_scores= cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
dtree_scores

array([0.96666667, 0.96666667, 0.9       , 0.96666667, 1.        ])

In [6]:
np.average(dtree_scores)

0.9600000000000002

In [7]:
#support vector machine (SVM)
svm_scores= cross_val_score(SVC(), iris.data, iris.target)
svm_scores

array([0.96666667, 0.96666667, 0.96666667, 0.93333333, 1.        ])

In [8]:
np.average(svm_scores)

0.9666666666666666

In [9]:
#random forest
rf_scores= cross_val_score(RandomForestClassifier(), iris.data, iris.target)
rf_scores

array([0.96666667, 0.96666667, 0.93333333, 0.96666667, 1.        ])

In [10]:
np.average(rf_scores)

0.9666666666666668

Logistic regression has yielded the highest score of 0.9733333333333334